### ISSUE TITLE
Build instructions launch Bash shell & Python REPL, requiring user intervention

### ISSUE THREAD

**Issue by alexjmacdonald** [grapl-security/grapl#347]
### What version of Grapl are you using?

Latest tag of v0.2.5.

### Have you tried reproducing the issue with the latest release?

Yes.

### Is this behavior happening in AWS, locally, or both?

Locally, haven't tested against AWS.

#### OS and OS version

Darwin 10.15.6

#### The output of `docker --version`

``` bash
Docker version 19.03.13, build 4484c46d9d
```

#### The output of `docker-compose --version`

``` bash
docker-compose version 1.27.4, build 40524192
```

#### The output of `docker image ls | grep grapl`

(this is as far as the build process got, I ran out of VM disk space for an unrelated reason)
``` bash
grapl/grapl-graphql-endpoint-build            latest              ba509c1ef15f        About a minute ago   346MB
grapl/grapl-notebook                          latest              d6df161646ba        2 minutes ago        1.68GB
grapl/grapl-model-plugin-deployer             latest              09316d30abcc        3 minutes ago        232MB
grapl/grapl-dgraph-ttl                        latest              3dedfb492f17        3 minutes ago        226MB
grapl/grapl-dynamodb-provision                latest              6ed709ccaa45        4 minutes ago        227MB
grapl/grapl-engagement-edge                   latest              6ed709ccaa45        4 minutes ago        227MB
grapl/grapl-graph-provision                   latest              6ed709ccaa45        4 minutes ago        227MB
grapl/grapl-engagement-creator                latest              6af4947b3303        4 minutes ago        233MB
grapl/grapl-analyzer-executor                 latest              6ef740628b39        4 minutes ago        227MB
grapl/analyzer-deployer                       latest              473f34af562e        4 minutes ago        220MB
grapl/grapl-python-deploy                     latest              f3583f990e59        4 minutes ago        112MB
grapl/grapl-model-plugin-deployer-build       latest              01ce45ecea40        5 minutes ago        618MB
grapl/grapl-dgraph-ttl-build                  latest              bcf84dca95f1        8 minutes ago        609MB
grapl/engagement-edge-build                   latest              5fffdb594d60        11 minutes ago       609MB
grapl/engagement-creator-build                latest              bf968303cbc9        17 minutes ago       619MB
grapl/analyzer-executor-build                 latest              6db2ad6bd38c        21 minutes ago       610MB
grapl/grapl-analyzerlib-python-build          latest              b93a33ed5c33        21 minutes ago       545MB
grapl/grapl-common-python-build               latest              b92c37b83bd1        21 minutes ago       533MB
grapl/grapl-graph-descriptions-python-build   latest              71da9477ca88        22 minutes ago       542MB
grapl/analyzer-deployer-build                 latest              c08c4f40340e        28 minutes ago       603MB
grapl/grapl-python-build                      latest              00f07732b0a8        29 minutes ago       451MB
grapl/grapl-sysmon-subgraph-generator         latest              84c1e8f69368        29 minutes ago       185MB
grapl/grapl-node-identifier-retry-handler     latest              25c3b049d3ad        29 minutes ago       190MB
grapl/grapl-node-identifier                   latest              73f7ec966e42        29 minutes ago       190MB
grapl/grapl-metric-forwarder                  latest              209fd9964d79        29 minutes ago       153MB
grapl/grapl-graph-merger                      latest              5017f3f2ef17        30 minutes ago       201MB
grapl/grapl-generic-subgraph-generator        latest              59130d1cf1ef        30 minutes ago       181MB
grapl/grapl-analyzer-dispatcher               latest              4ebfb4c1f8ac        30 minutes ago       174MB
grapl/grapl-rust-src-build                    latest              dabbf725b614        35 minutes ago       9.18GB
```

### Steps to reproduce the issue

Build command as offered in `BUILDING.md`, modified for (a) Fish shell and (b) repo-local `dobi` binary:
`env TAG=latest GRAPL_RELEASE_TARGET=debug ./dobi --no-bind-mount build`

### Expected behavior

Build finishes without any user interaction beyond initial build command.

### Observed behavior

The build process dropped me into a Bash shell at first, and then several Python REPLs, all within grapl containers.

At first, I was confused and hit CTRL-D to exit, which resulted in an apparent stall for several seconds -- though upon retrying, it appears that the Docker container was actually performing build steps.

```
Step 88/88 : RUN mkdir -p /home/grapl/dist &&     cp target/x86_64-unknown-linux-musl/${release_target}/analyzer-dispatcher dist/analyzer-dispatcher &&     cp target/x86_64-unknown-linux-musl/${release_target}/generic-subgraph-generator dist/generic-subgraph-generator &&     cp target/x86_64-unknown-linux-musl/${release_target}/graph-merger dist/graph-merger &&     cp target/x86_64-unknown-linux-musl/${release_target}/metric-forwarder dist/metric-forwarder &&     cp target/x86_64-unknown-linux-musl/${release_target}/node-identifier dist/node-identifier &&     cp target/x86_64-unknown-linux-musl/${release_target}/node-identifier-retry-handler dist/node-identifier-retry-handler &&     cp target/x86_64-unknown-linux-musl/${release_target}/sysmon-subgraph-generator dist/sysmon-subgraph-generator
 ---> Running in 9e71cb2a9132
Removing intermediate container 9e71cb2a9132
 ---> dabbf725b614
Successfully built dabbf725b614
Successfully tagged grapl/grapl-rust-src-build:latest
[image:build rust-build] grapl/grapl-rust-src-build Created
[job: build-rust] ./dist/ Start
Step 1/2 : FROM grapl/grapl-rust-src-build:latest
 ---> dabbf725b614
Step 2/2 : COPY ./dist /home/grapl/dist
 ---> a347cca37be6
Successfully built a347cca37be6
Successfully tagged grapl/grapl-rust-src-build:job-grapl-alex-build-rust
grapl@17bfc1cd1c7b:~$ ls
Cargo.lock  analyzer-dispatcher  dist                        graph-descriptions   graph-merger  grapl-observe     node-identifier            target
Cargo.toml  derive-dynamic-node  generic-subgraph-generator  graph-generator-lib  grapl-config  metric-forwarder  sysmon-subgraph-generator  vendor
grapl@17bfc1cd1c7b:~$ exit


[job: build-rust] ./dist/ Done
[image:build rust-build] grapl/grapl-rust-src-build is fresh
Step 1/5 : FROM alpine AS grapl-analyzer-dispatcher
```

```
[image:build analyzer-deployer-build] grapl/analyzer-deployer-build Created
[job: build-analyzer-deployer] ./dist/analyzer-deployer/lambda.zip Start
Step 1/2 : FROM grapl/analyzer-deployer-build:latest
 ---> c08c4f40340e
Step 2/2 : COPY ./dist /home/grapl/dist
 ---> bcb9b111b394
Successfully built bcb9b111b394
Successfully tagged grapl/analyzer-deployer-build:job-grapl-alex-build-analyzer-deployer

Python 3.7.9 (default, Sep 10 2020, 17:21:15)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>

[job: build-analyzer-deployer] ./dist/analyzer-deployer/lambda.zip Done
[image:build python-build] grapl/grapl-python-build is fresh
[image:build python-build] grapl/grapl-python-build is fresh
Step 1/8 : FROM grapl/grapl-python-build:latest AS grapl-graph-descriptions-python-build
 ---> 00f07732b0a8
```

```
[image:build analyzer-executor-build] grapl/analyzer-executor-build Created
[job: build-analyzer-executor] ./dist/analyzer-executor/lambda.zip Start
Step 1/2 : FROM grapl/analyzer-executor-build:latest
 ---> 6db2ad6bd38c
Step 2/2 : COPY ./dist /home/grapl/dist
 ---> 7a1ca173cf54
Successfully built 7a1ca173cf54
Successfully tagged grapl/analyzer-executor-build:job-grapl-alex-build-analyzer-executor
Python 3.7.9 (default, Sep 10 2020, 17:21:15)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
[job: build-analyzer-executor] ./dist/analyzer-executor/lambda.zip Done
[image:build python-build] grapl/grapl-python-build is fresh
[image:build python-build] grapl/grapl-python-build is fresh
```

**Comment by alexjmacdonald** [comment]
Between the `Dockerfile`s and `dobi.yaml`, it appears there isn't an `ENTRYPOINT` or `CMD` command.  It appears that, on my machine, it defaults to running `/bin/sh` as the command for the Rust build image (& Python for the Python build images).  `/bin/sh` seems like an expectable default, though I don't know how it's running Python for the Python build images.

In any case, I was able to resolve the issue locally by putting a no-op command (`echo build is done lol` worked) as a `command` parameter to each build job:

```
# modified dobi.yaml
job=build-analyzer-deployer:
  use: analyzer-deployer-build
  mounts:
    - dist
  artifact:
    - ./dist/analyzer-deployer/lambda.zip
  command: echo py is done lol # no-op
```

```
# resulting output, no REPL
[image:build analyzer-deployer-build] grapl/analyzer-deployer-build Created
[job: build-analyzer-deployer] echo py is done lol -> ./dist/analyzer-deployer/lambda.zip Start
Step 1/2 : FROM grapl/analyzer-deployer-build:latest
 ---> f3812b9e9cd5
Step 2/2 : COPY ./dist /home/grapl/dist
 ---> ba096f497690
Successfully built ba096f497690
Successfully tagged grapl/analyzer-deployer-build:job-grapl-alex-build-analyzer-deployer
py is done lol
[job: build-analyzer-deployer] echo py is done lol -> ./dist/analyzer-deployer/lambda.zip Done
```

**Comment by alexjmacdonald** [comment]
Ok, a little more investigation, and I found that the python dockerfiles are using FROM: `https://github.com/docker-library/repo-info/blob/master/repos/python/remote/3.7-slim-buster.md`, which has a default command of `python3`.

It appears that the root cause of a REPL being launched is that the default command of `python3` is being executed.

**Comment by alexjmacdonald** [comment]
Closed by #355.
